src/core/toga/widgets/selection.py

```diff
@@ -60,7 +59,7 @@ def value(self):
 
     @value.setter
     def value(self, value):
-        if value not in self._items:
+        if value not in (i.field for i in self.items):
```

Comment:
We should also account for the case where the user provides the actual row object (whatever the underlying item Source is storing) as the selected value. Likewise on the getter; we should be returning the underlying row object, not the display label.